docs/posts/querying-pypi-metadata-compiled-languages/index.qmd

```diff
@@ -0,0 +1,267 @@
+---
+title: Querying every file in every release on the Python Package Index (redux)
+author: Gil Forsyth
+date: 2023-11-15
+categories:
+    - blog
+---
+
+Seth Larson wrote a great [blog
+post](https://sethmlarson.dev/security-developer-in-residence-weekly-report-18)
+on querying a PyPI dataset to look for trends in the use of memory-safe
+languages in Python.
+
+You should check out Seth's article for more information on the dataset (and
+it's a good read!) but it caught our eye because it makes use of
+[DuckDB](https://duckdb.org/) to clean the data for analysis.
+
+That's right up our alley here in Ibis land, so let's see if we can duplicate
+Seth's results (and then continue on to plot them!)
+
+## Grab the data (locations)
+
+Seth showed (and then safely decomposed) a nested `curl` statement and that's
+always viable -- we're in Python land so why not grab the filenames using
+`urllib3`?
+
+```{python}
+import urllib3
+
+http = urllib3.PoolManager()
+
+resp = http.request("GET", "https://github.com/pypi-data/data/raw/main/links/dataset.txt")
+
+parquet_files = resp.data.decode().split()
+parquet_files
+```
+
+## Grab the data
+
+Now we're ready to get started with Ibis!
+
+DuckDB is clever enough to grab parquet metadata.  We can use `read_parquet` to
```

Comment:
> DuckDB is clever enough to grab parquet metadata.

Hm, I'm not sure I follow why this is important here. Maybe "grab parquet metadata" needs more explanation?

Reply:
yeah, I'll expand on this, I was aiming for "you don't need to download everything beforehand"